src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -388,7 +391,11 @@ private Plan.KeysIteration buildKeysIterationPlan()
         if (orderer != null)
             keysIterationPlan = planFactory.sort(keysIterationPlan, orderer);
 
-        assert keysIterationPlan != planFactory.everything; // This would mean we have no WHERE nor ANN clauses at all
+        // This would mean we have no WHERE nor ANN clauses at all; this can happen in case an index was dropped after the
+        // query was initiated
+        if (keysIterationPlan == planFactory.everything)
+            throw invalidRequest(INDEX_MAY_HAVE_BEEN_DROPPED + StatementRestrictions.REQUIRES_ALLOW_FILTERING_MESSAGE);
```

Comment:
Nit: we need a space between the two error messages. It can be added here or at the end of the constant value of `INDEX_MAY_HAVE_BEEN_DROPPED`.

Reply:
Actually, we can do the concatenation in the declaration of `INDEX_MAY_HAVE_BEEN_DROPPED`:
```
public static final String INDEX_MAY_HAVE_BEEN_DROPPED = "An index may have been dropped. " +
                                                         StatementRestrictions.REQUIRES_ALLOW_FILTERING_MESSAGE;
```